### ISSUE TITLE
Preparation of Custom Training Data

### ISSUE THREAD

**Issue by danurag460** [Extreme-classification/deepxml#17]
Hi Team,
I was using DeepXML framework and not able to understand that how to prepare train/test data for my custom dataset.
I have a set of label text documents, short text documents and their corresponding labels.

Can you please answer how to prepare trn_X_Xf.txt, trn_X_Y.txt, tst_X_Xf.txt, tst_X_Y.txt and fasttextB_embeddings_300d.npy files?

what does 40000 number says?

Thanks!

**Comment by kunaldahiya** [comment]
Hi,

1. For split creation: you can first try randomly choosing 80% of points as train and rest would become validation/test set. 
2. Given a raw text file and labels, here is an to create tf-idf features [example](https://github.com/kunaldahiya/pyxclib/blob/master/xclib/examples/sparse_bow_features_from_raw_data.py) (with unigram tokens). You can either split the features and labels using the perl script given in documentation or dump them directly as separate files (use [this function](https://github.com/kunaldahiya/pyxclib/blob/985421a30a80380e172ca145399eae58134ae3f3/xclib/data/data_utils.py#L81) in place of 'write_data'). 40K is the number of tokens in the vocabulary.
3.  Once you have the vocabulary you can get the 'fasttextB_embeddings_300d.npy' file using this [example](https://github.com/kunaldahiya/pyxclib/blob/master/xclib/examples/get_ftx_embeddings.py).


Please note that if your task has label text then I suggest the [NGAME algorithm](https://github.com/Extreme-classification/ngame) which is based on a transformer architecture and can make effective use of label-text.
